### PR TITLE
Fix case sensitivity bug on finding `dockerfile`

### PR DESCRIPTION
**Pull request by 733amir**

### CODE REVIEW

File: fandogh_cli/workspace.py [hunk 1]
```diff
@@ -9,7 +9,7 @@ def __init__(self, workspace_config=None):
         workspace_config = workspace_config or {}
         self.path = workspace_config.get('path', os.getcwd())
         self.zip_file_name = os.path.join(self.path, 'workspace.zip')
-        files = os.listdir(self.path)
+        files = [file.lower() for file in os.listdir(self.path)]
```
Comment (s-soroosh) [inline fandogh_cli/workspace.py]: Hi @733amir, 
Wonder why we need to lower case all file names here?

Comment (733amir) [reply fandogh_cli/workspace.py]: Hi @psycho-ir,
To get case insensitive search of `dockerfile` and other files and directories in the future.

Comment (MahdiZareie) [reply fandogh_cli/workspace.py]: what about : 
```
# at the top
import sys
if sys.version[0] == '2':
 from itertools import imap as map

# in here
self.has_docker_file = 'dockerfile' in map(str.lower, files)
```

Comment (733amir) [reply fandogh_cli/workspace.py]: Good! It is more readable now.
What about `.dockerignore` file? shouldn't we consider that too?

```python
files = map(str.lower, os.listdir(self.path))
self.has_docker_ignore = '.dockerignore' in files
self.has_docker_file = 'dockerfile' in files
```

File: fandogh_cli/image_commands.py [hunk 2]
```diff
@@ -111,7 +111,7 @@ def publish(version, detach):
             return
     workspace = Workspace()
 
-    if not workspace.has_docker_file:
+    if not workspace.docker_file_name:
```
Comment (s-soroosh) [inline fandogh_cli/image_commands.py:114]: let's have 2 separated changeset. 
one for dockerignore bug and the other for dockerignore improvement.

Regarding the bug it seems to me we only need to change the `has_docker_ignore` logic to sth like:
`self.has_docker_file = 'dockerfile' in map(str.lower, files)`

do you have time to test it? I believe it's enough. 
once we merge it we can improve the docker ignore part.

File: fandogh_cli/workspace.py (hunk 1 above)
Comment (MahdiZareie) [reply fandogh_cli/workspace.py]: I think we should limit the scope of this PR to Dockerfile case, we can work on .dockerignore in another PR.

**Review by MahdiZareie — CHANGES_REQUESTED** [review]
Please reduce the PR changes to limit it on  `self.has_docker_file` logic, then we can merge it.

**Comment by EAliakbar** [comment]
why we can't use something like this?
`self.has_docker_file = 'Dockerfile' in files or 'dockerfile' in files`
`docker build` only search for these two files when `-f` switch is not specified

**Comment by s-soroosh** [comment]
> why we can't use something like this?
> `self.has_docker_file = 'Dockerfile' in files or 'dockerfile' in files`
> `docker build` only search for these two files when `-f` switch is not specified

sure, we can do it in this way as well.

**Pull request closed without merging**